docs/client_reference.rst

```diff
@@ -291,12 +295,40 @@ The client session supports the context manager protocol for self closing.
       :param bool expect100: Expect 100-continue response from server.
                              ``False`` by default (optional).
 
-      :param bool raise_for_status: Automatically call :meth:`ClientResponse.raise_for_status()` for
-                                    response if set to ``True``.
-                                    If set to ``None`` value from ``ClientSession`` will be used.
-                                    ``None`` by default (optional).
+      :param ~typing.Union[bool, callable] raise_for_status:
+
+         Automatically apply a checker for failed status code (usually a code
+         that is greater than ``400``).
+
+         ``None`` (default) means that ``raise_for_status`` argument of
+         :class:`ClientSession` constructor controls this behavior.
+
+         Set the parameter to ``True`` if you need to enforce
+         :meth:`ClientResponse.raise_for_status` call.
+
+         Set the argument to ``False`` to suppress a HTTP status chacker even if
+         :class:`ClientSession` enables it.
+
+         Use *async callback* to call user code for that accepts a
+         :class:`ClientResponse` and raises an exception to prevent the future
+         processing.
+
+         The following callback example is a functional equivalent if
+         ``raise_for_status=True``::
+
+             async def custom_check(response):
+                 if 400 <= response.status:
+                     raise aiohttp.ClientResponseError(
```

Comment:
looks like the indentations is wrong here, but that might just be how github is displaying this.

Reply:
Heh, looks like the indentation is broken even in 3.6 release: http://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientSession

I'll fix it in a separate PR